DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -43,59 +53,62 @@ internal class RecordingCoordinator {
         recorder: Recording,
         sampler: Sampler,
         telemetry: Telemetry,
-        startRecordingImmediately: Bool,
-        methodCallTelemetrySamplingRate: Float = 0.1
-    ) {
+        recordingTrigger: RecordingTriggering,
+        methodCallTelemetrySamplingRate: Float = 0.1,
+        dateProvider: DateProvider = SystemDateProvider(),
+        queue: Queue = MainAsyncQueue()
```

Comment:
/suggestion To prevent test flakiness in the future, it might be best to remove the default `queue` value and let the feature inject the instance. This way, we are forced to specify a queue in tests.